tests/integration/targets/test_proxysql_info/tasks/main.yml

```
@@ -0,0 +1,32 @@
+---
+- name: "{{ role_name }} | proxysql_info | test in check_mode"
+  check_mode: yes
+  community.proxysql.proxysql_info:
+    login_user: admin
+    login_password: admin
+  register: proxysql_information
+
+- debug:
+    var: proxysql_information.version
+
+- name: verify some return values
+  assert:
+    that: 
+      - "'global_variables' in proxysql_information.tables"
+      - "'mysql_servers' in proxysql_information.tables"
+      - "'runtime_mysql_servers' in proxysql_information.tables"
+      - proxysql_information.version.major == 2
+
+- name: "{{ role_name }} | proxysql_info | test not in check_mode"
+  community.proxysql.proxysql_info:
+    login_user: admin
+    login_password: admin
+  register: proxysql_information
+
+- name: verify some return values
+  assert:
+    that: 
+      - "'global_variables' in proxysql_information.tables"
```

```suggestion
      - "'global_variables' in proxysql_information.tables"
```
can we check tables a bit more deeply? Especially the content of the global_variables? If we can check for some usual default values, it would be perfect.

---

Yes, it should be also possible to remove all `shell:` usages in the integration test with `proxysql_info`.  
https://github.com/ansible-collections/community.proxysql/blob/main/tests/integration/targets/test_proxysql_query_rules/tasks/base_test.yml#L55  
But that's something for a separate PR imo.

---

yeah, sounds sensible